### PR TITLE
Vault Timeout Policy

### PR DESCRIPTION
**Pull request by Hinton**
## Objective
Implements a *Vault Timeout Policy* which lets organizations restrict the maximum vault timeout. It also adds the new `VaultTimeoutInputComponent` which supports `Custom` timeouts.

### Code Changes
- **bitwarden_license/src/app/app.component.ts**: Add `AppComponent` for bitwarden_licensed version. Which adds the `MaximumVaultTimeoutPolicy` to policies list.
- **bitwarden_license/src/app/app.module.ts**: Add `ReactiveFormsModule`, declare new `AppModule` and `MaximumVaultTimeoutPolicyComponent`.
- **bitwarden_license/src/app/policies/maximum-vault-timeout.component.{ts,html}**: New policy for Vault Timeout.
- **src/app/app.module.ts**: Move `AppComponent` from `OssModule` to `AppModule`.
- **src/app/organizations/policies/base-policy.component.ts**: Refactor `BasePolicy` to have two new functions `loadData` and `buildRequestData`. Which helps in minimizing the duplicated code in policies.
- **src/app/settings/options.component.{ts,html}**: Use the new vault timeout input component.
- **src/services/webPlatformUtils.service.ts**: Remove deprecated `lockTimeout`.

### Screenshots
image
image

Depends on https://github.com/bitwarden/jslib/pull/474

### CODE REVIEW

File: bitwarden_license/src/app/policies/maximum-vault-timeout.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<app-callout type="tip" title="{{'prerequisite' | i18n}}">
+    {{'requireSsoPolicyReq' | i18n}}
```
Comment (MGibson1) [inline bitwarden_license/src/app/policies/maximum-vault-timeout.component.html:2]: These strings seem poorly named now. It's not really SSO, it's just single org

File: src/locales/en/messages.json [hunk 2]
```diff
@@ -4186,5 +4186,39 @@
   },
   "updateMasterPasswordWarning": {
     "message": "Your Master Password was recently changed by an administrator in your organization. In order to access the vault, you must update your Master Password now. Proceeding will log you out of your current session, requiring you to log back in. Active sessions on other devices may continue to remain active for up to one hour."
+  },
+  "maximumVaultTimeout": {
+    "message": "Vault Timeout"  
+  },
+  "maximumVaultTimeoutDesc": {
+    "message": "Configure an automatic vault timeout for all users based on a maximum duration."
```
Comment (MGibson1) [inline src/locales/en/messages.json:4194]: Isn't something like "Configure a maximum automatic vault timeout for all users." closer?

I don't believe we're setting the vault timeout TO the provided value for all user, just requiring timeout to be at most the given value.

File: src/locales/en/messages.json [hunk 3]
```diff
@@ -4186,5 +4186,39 @@
...
+        "content": "$2",
+        "example": "5"
+      }
+    }
+  },
+  "customVaultTimeout": {
+    "message": "Custom Vault Timeout"
+  },
+  "vaultTimeoutToLarge": {
+    "message": "Your vault timeout exceeds the restrictions set by your organization."
```
Comment (MGibson1) [inline src/locales/en/messages.json:4222]:
```
Your vault timeout exceeds the restriction set by your organization.
```
 or
```
Your vault timeout exceeds restrictions set by your organization.
```

**Review by MGibson1 — COMMENTED** [review]
This looks good. Just a couple of messaging string suggestions.

You'll also still need to update jslib to point at master (unless you've sneakily done that). I think this is pointing at your feature branch, right now.

File: bitwarden_license/src/app/policies/maximum-vault-timeout.component.html (hunk 1 above)
Comment (Hinton) [reply bitwarden_license/src/app/policies/maximum-vault-timeout.component.html:2]: If we rename it, all locales needs to be updated which feels unnecessary.

File: src/locales/en/messages.json (hunk 2 above)
Comment (Hinton) [reply src/locales/en/messages.json:4206]: I agree it's weirdly phrased, it's taken from the Asana ticket, but I'll check if we can change it.

File: bitwarden_license/src/app/policies/maximum-vault-timeout.component.html (hunk 1 above)
Comment (MGibson1) [reply bitwarden_license/src/app/policies/maximum-vault-timeout.component.html:2]: You're right, but we need to be able to have these evolve over time.

I made a quick PR to solve this problem here: https://github.com/bitwarden/jslib/pull/484

The usage would be:

`messages.json:`
```
{
  ...
  "requireSingleOrgPolicy" : {
    "message": "..."
    ...
    "previousNames": [
      "requireSsoPolicyReq"
    ]
  }
  ...
}
```

**Review by MGibson1 — DISMISSED** [review]
Looks good, up to you if you want to wait on bitwarden/jslib#484

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**